### DISCUSSION TITLE
Dynamic theme switching

### DISCUSSION THREAD

**Discussion by luicfrr** [tinymce/tinymce#8413] [General]
Is it possible to dynamic switch between light/dark themes?
I'm using React and I could find nothing in the project about inline styling, dynamic theming, switching editor main color, etc.. based on a variable controled with react state.

Is there any way to dynamically switch themes with this text editor? If positive, how to do this?

**Comment by lostkeys** [comment]
Hi @nonam4, here is the documentation for customizing the UI and editor skins: https://www.tiny.cloud/docs/tinymce/6/customize-ui/
That page contains an example showing how to switch between light and dark.

— Fredrik, Product Manager

**Reply by jweaver0312** [reply]
While it does switching between light and dark, it is not necessarily truly 100% dynamic. In my opinion, true 100% dynamic would allow it to switch between light and dark in sync with the device. Currently, with the documentation and my testing, it only flows with device preference upon loading the script only.

For example, if it loads with dark theme and I chanage the device to light theme, TinyMCE does not follow suit and change to light theme with the device changing over (without reloading the page)

**Reply by luicfrr** [reply]
thanks, I've managed to use it with non dynamic light/dark theme switching

**Reply by lostkeys** [reply]
I haven't had time to test this myelf but what happens if you create a content_css that uses [@import with a mediaquery](https://developer.mozilla.org/en-US/docs/Web/CSS/@import#importing_css_rules_conditionally) to load either the light or dark skin. I wonder if the browser would "know" to switch automatically without a reload. If anyone tries, let us know!

**Comment by TheSpyder** [comment]
At the end of the day any dynamic theming is difficult to achieve without using CSS variables - which would be a fine option today, but wasn't when we last rewrote the UI styles in 2018. So long as we're using `less` to compile the stylesheets this there isn't really a nice way to achieve this.

**Comment by lostkeys** [comment]
Hold my beer...Here is a proof of concept people can explore https://fiddle.tiny.cloud/i1iaab (it worked on my machine)
